src/rust/src/demuxer/demux.rs

```diff
@@ -0,0 +1,726 @@
+#[cfg(windows)]
+use std::os::windows::io::IntoRawHandle;
+use std::path::Path;
+use std::ptr::{null, null_mut};
+
+cfg_if! {
+    if #[cfg(test)] {
+        use crate::demuxer::demux::tests::{print_file_report, start_tcp_srv, start_upd_srv};
+    } else {
+        use crate::{print_file_report, start_tcp_srv, start_upd_srv};
+        #[cfg(feature = "enable_ffmpeg")]
+        use crate::init_ffmpeg;
+    }
+}
+
+impl CcxDemuxer<'_> {
+    pub fn get_filesize(&mut self) -> i64 {
+        // Get the file descriptor from ctx.
+        let in_fd = self.infd;
+        if in_fd < 0 {
+            return -1;
+        }
+
+        // SAFETY: We are creating a File from an existing raw fd.
+        // To prevent the File from closing the descriptor on drop,
+        // we call into_raw_fd() after using it.
+        #[cfg(unix)]
+        let mut file = unsafe { File::from_raw_fd(in_fd) };
+        #[cfg(windows)]
+        let mut file = open_windows(in_fd);
+
+        // Get current position: equivalent to LSEEK(in, 0, SEEK_CUR);
+        let current = match file.stream_position() {
+            Ok(pos) => pos,
+            Err(_) => {
+                // Return the fd back and then -1.
+                self.drop_fd(file);
+                return -1;
+            }
+        };
+
+        // Get file length: equivalent to LSEEK(in, 0, SEEK_END);
+        let length = match file.seek(SeekFrom::End(0)) {
+            Ok(pos) => pos,
+            Err(_) => {
+                self.drop_fd(file);
+                return -1;
+            }
+        };
+
+        #[allow(unused_variables)]
+        let ret = match file.seek(SeekFrom::Start(current)) {
+            Ok(pos) => pos,
+            Err(_) => {
+                self.drop_fd(file);
+                return -1;
+            }
+        };
+
+        // Return the fd back to its original owner.
+        self.drop_fd(file);
+        length as i64
+    }
+
+    pub fn reset(&mut self) {
+        {
+            self.startbytes_pos = 0;
+            self.startbytes_avail = 0;
+            self.num_of_pids = 0;
+            // Fill have_pids with -1 for (MAX_PSI_PID+1) elements.
+            let len_have = MAX_PSI_PID + 1;
+            if self.have_pids.len() < len_have {
+                self.have_pids.resize(len_have, -1);
+            } else {
+                self.have_pids[..len_have].fill(-1);
+            }
+            // Fill pids_seen with 0 for MAX_PID elements.
+            if self.pids_seen.len() < MAX_PID {
+                self.pids_seen.resize(MAX_PID, 0);
+            } else {
+                self.pids_seen[..MAX_PID].fill(0);
+            }
+            // Set each min_pts[i] to u64::MAX for i in 0..=MAX_PSI_PID.
+            for i in 0..=MAX_PSI_PID {
+                if !self.min_pts.is_empty() {
+                    self.min_pts[i] = u64::MAX;
+                }
+            }
+            // Fill stream_id_of_each_pid with 0 for (MAX_PSI_PID+1) elements.
```

Comment:
You are actually not filling min_pts if it is empty whilst C version always fill it. This should fix it:

```suggestion
            if self.min_pts.len() != MAX_PSI_PID + 1 {
                self.min_pts.resize(MAX_PSI_PID + 1, u64::MAX);
            } else {
                // Set each min_pts[i] to u64::MAX for i in 0..=MAX_PSI_PID.
                for i in 0..=MAX_PSI_PID {
                    self.min_pts[i] = u64::MAX;
                }
            }
            // Fill stream_id_of_each_pid with 0 for (MAX_PSI_PID+1) elements.
```